### PR TITLE
feat(unicorn): api is a part of cocaine now

### PR DESCRIPTION
**Pull request by antmat**
Moved it as is.
Give me a second to apply cocaine-core code style.

### CODE REVIEW

**Comment by antmat** [comment]
done

File: include/cocaine/unicorn/value.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+*/
+
+#ifndef COCAINE_UNICORN_VALUE
+#define COCAINE_UNICORN_VALUE
+
+#include "cocaine/common.hpp"
+#include "cocaine/dynamic.hpp"
+#include "cocaine/traits/dynamic.hpp"
+
+#include <msgpack.hpp>
```
Comment (3Hren) [inline include/cocaine/unicorn/value.hpp]: Why?

File: include/cocaine/unicorn/value.hpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace cocaine { namespace unicorn {
+
+typedef long long version_t;
+typedef cocaine::dynamic_t value_t;
+
+static constexpr version_t MIN_VERSION = -2;
```
Comment (3Hren) [inline include/cocaine/unicorn/value.hpp]: Style.

File: include/cocaine/unicorn/value.hpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#define COCAINE_UNICORN_VALUE
+
+#include "cocaine/common.hpp"
+#include "cocaine/dynamic.hpp"
+#include "cocaine/traits/dynamic.hpp"
+
+namespace cocaine { namespace unicorn {
+
+typedef long long version_t;
+typedef cocaine::dynamic_t value_t;
```
Comment (3Hren) [inline include/cocaine/unicorn/value.hpp]: Rewrite with forwards instead.

File: include/cocaine/api/unicorn.hpp [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+*/
+
+#ifndef COCAINE_UNICORN_API_HPP
+#define COCAINE_UNICORN_API_HPP
+
+#include <cocaine/repository.hpp>
+
+#include <cocaine/forwards.hpp>
+
+#include <cocaine/unicorn/path.hpp>
```
Comment (3Hren) [inline include/cocaine/api/unicorn.hpp]: FORWARDS

File: include/cocaine/api/unicorn.hpp [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+#include <cocaine/unicorn/value.hpp>
```
Comment (3Hren) [inline include/cocaine/api/unicorn.hpp]: FORWARDS TOO

File: include/cocaine/api/unicorn.hpp [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+#include "cocaine/unicorn/writable.hpp"
```
Comment (3Hren) [inline include/cocaine/api/unicorn.hpp]: AND HERE

File: include/cocaine/unicorn/path.hpp [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_PATH
+#define COCAINE_UNICORN_PATH
+
+#include <string>
```
Comment (3Hren) [inline include/cocaine/unicorn/path.hpp]: BUT NOT HERE

File: include/cocaine/unicorn/writable.hpp [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_WRITABLE_ADAPTER_HPP
+#define COCAINE_UNICORN_WRITABLE_ADAPTER_HPP
+
+#include <cocaine/rpc/dispatch.hpp>
```
Comment (3Hren) [inline include/cocaine/unicorn/writable.hpp]: To include dispatch is a crime? Are you criminal scum? https://www.youtube.com/watch?v=yieYXj5cDKE

File: include/cocaine/api/unicorn.hpp [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+        synchronized<std::map<std::string, std::weak_ptr<unicorn_t>>> instances;
+    };
+};
+
+category_traits<unicorn_t>::ptr_type
+unicorn(context_t& context, const std::string& name);
+
+}}
+
+#endif // COCAINE_UNICORN_API_HPP
```
Comment (3Hren) [inline include/cocaine/api/unicorn.hpp]: Teach your IDE to add empty line at the end of a file.

**Pull request merged**